test/e2e/handler/nns_update_timestamp_test.go

```diff
@@ -28,24 +24,49 @@ var _ = Describe("[nns] NNS LastSuccessfulUpdateTime", func() {
 			originalNNSs[node] = nodeNetworkState(key)
 		}
 	})
-	Context("when network configuration hasn't change", func() {
-		It("should not be updated", func() {
+	Context("when network configuration is not changed by a NNCP", func() {
+		It("should be updated only if configuration changed", func() {
+			// Give enough time for the NNS to be reconcile(2 interval times)
+			timeToWait := 2 * nmstatenode.NetworkStateRefresh
+			for retry := 4; retry > 0; retry-- {
+				By(fmt.Sprintf("Waiting %s to check for NNS refresh timestamp", timeToWait))
+				time.Sleep(timeToWait)
+				for node, originalNNS := range originalNNSs {
+					key := types.NamespacedName{Name: node}
+					currentStatus := nodeNetworkState(key).Status
+					originalStatus := originalNNS.Status
+					if currentStatus.CurrentState.String() == originalStatus.CurrentState.String() {
+						By(fmt.Sprintf("Check LastSuccessfulUpdateTime changed at %s", node))
+						Expect(currentStatus.LastSuccessfulUpdateTime).To(Equal(originalStatus.LastSuccessfulUpdateTime))
+						retry = 0
+					} else {
+						By("Check LastSuccessfulUpdateTime again network configuration changed")
```

Comment:
Let's hope nothing would change the networking too often, still, we may want to fail here if we run out or attemts don't we?

Reply:
Refactored using eventually playing with the fact that if we use an interval of 2 * refresh and a timeout of 4 * interval we have the same behaviour and also fails on retry limit.